dandi/dandiapi.py

```diff
@@ -1071,6 +1071,20 @@ def iter_upload_raw_asset(
             self, metadata=asset_metadata, jobs=jobs, replacing=replace_asset
         )
 
+    def is_nwb(self) -> bool:
```

Comment:
@jwodder I feel that we had a similar case somewhere where I have pushed for replacing `is_nwb` and `is_zarr` with something like `format` property .  Do you remember what was it? 
@bendichter I think that situation is similar here and we should not overfit for NWB here and needing to grow API for any type of file formats which might be in the dandiset.  Moreover, `is_nwb` is IMHO a suboptimal name. if anything should be `has_nwb` since dandiset is not of NWB format.

What if there is a more general `has_any(format: str)` where it would understand `format` *as-is* if `.startswith("RRID")` and otherwise knows a few mappings, such as the used RRIR for `NWB`, and may be right away one for zarr if we know of any (@satra)?

This was we would not need to expand API and duplicate code to support more "formats"

Reply:
@yarikoptic I think you're thinking of the discussion that lead to [this commit](https://github.com/dandi/dandi-cli/commit/c0a32efff54c34e2b79c2d551def3a396a2e1fb6).

Reply:
what would be my new syntax for determining if a dandiset contains NWB files?

Reply:
@jwodder -- thank you! Indeed it is `BaseRemoteAssset.asset_type`  https://github.com/dandi/dandi-cli/blob/master/dandi/dandiapi.py#L1353 I had in mind.  After starting to craft a patch for @bendichter I realized that it allowed to differentiate at different "level" -- only `blob` vs `zarr`, not at the "higher" (if not orthogonal) level of the asset format or "mime" type. `blob` could be a nifti or nwb or any other of hundreds of mimetypes or in other words `dataStandard`s as we call that in our assets summary which @bendichter is to use here.

So I think it could be something like (prototyping to expedite since not sure what part of "syntax" you had in mind @bendichter  ;) ):

```
    def has_data_standard(data_standard: str) -> bool:
        """
        Returns True if the Dandiset contains one or more assets conformant to a given type.
    
       `data_standard` should be an `RRID:` identifier, or a string for one of the known data standards
       (ATM known only `nwb`). Code then checks for the corresponding RRID to be listed in the
       "dataStandard" field of the assetsSummary of the dandiset.
        """
        known = {
            'nwb': "RRID:SCR_015242",
        }
        if not data_standard.startswith('RRID:'):
                if data_standard.lower() in known_types:
                    data_standard = known_types[data_standard.lower()]
                else:
                    raise ValueError(f"Do not know about data standard {data_standard}. Provide an RRID or one of known: {', '.join(known)}")
        assetsSummary = self.get_raw_metadata()["assetsSummary"]
        if "dataStandard" not in assetsSummary:
            return False
        return any(
            x["identifier"] == data_standard for x in assetsSummary["dataStandard"]
        )
```
    
I also wonder if we should discriminate the use case of `dataStandard` being not populated and return None, so code outside could tell apart the case where such is not populated. Code outside would still be able to use it as a `bool`.  But not sure if worth it.

Reply:
I meant how would I use this function as a user. So it would be `remote_dandiset.has_data_standard("nwb")`? That works for me.

I'm fine with returning `None` if no data standard is defined, though I probably wouldn't use that feature.

Reply:
> `remote_dandiset.has_data_standard("nwb")`? That works for me.

yes, that is what I thought about. @jwodder -- do you see a better naming/signature?  What do you think about `None` for return?

Reply:
@yarikoptic I have no opinion on the name.  I think a lack of `dataStandard` should result in an exception, as it's an exceptional condition.

Reply:
@jwodder I think that would require me to use this method within a try/except which I'd rather not do.

Reply:
ok, looking at the schema and code

```shell
$> git grep -p dataStand
dandischema/metadata.py=def _add_asset_to_stats(assetmeta: Dict[str, Any], stats: _stats_type) -> None:
dandischema/metadata.py:    stats["dataStandard"] = stats.get("dataStandard", [])
dandischema/metadata.py:        if models.nwb_standard not in stats["dataStandard"]:
dandischema/metadata.py:            stats["dataStandard"].append(models.nwb_standard)
dandischema/metadata.py:        if models.bids_standard not in stats["dataStandard"]:
dandischema/metadata.py:            stats["dataStandard"].append(models.bids_standard)
dandischema/models.py=class AssetsSummary(DandiBaseModel):
dandischema/models.py:    dataStandard: Optional[List[StandardsType]] = Field(readOnly=True)
```
dataStandard is optional and populated only if there is any asset, so empty dandisets do not have it , as verified on `curl -X GET "https://api.dandiarchive.org/api/dandisets/000233/versions/draft/" -H  "accept: application/json" `, so I think it would not be an exceptional condition. Ok, let's keep it just to `False` in case of its absence.

Reply:
@bendichter what are your plans on this PR - should we generalize it the way I proposed or some other way, or forget/close for now?

Reply:
`has_any` is too wide semantically for me, and it should not default to `NWB` IMHO, so what about suggested `has_data_standard(data_standard: str)` ? ;-)

Reply:
@yarikoptic ok I can adjust this PR to that syntax